nodejs/device/src/duration.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+export function msToDurationString(delta: number) {
+  // [D:]h:m:s[.UUUUUU]
+  delta = Math.trunc(delta);
+  let [ms, sec, min, hr] = [1000, 60, 60, 24].map((currentValue) => {
+    const mod = delta % currentValue;
+    delta = (delta - mod) / currentValue;
+    return mod;
+  });
```

Comment:
I appreciate the way you're trying to do this, and I think this kind of thinking can lead to some beautiful code, but this is a bit confusing:
1. `delta` and `mod` mean something mathematically, but mapping them to reality is difficult.  I think this is because the units change on every iteration.  `currentValue` is also confusing because it's not really the current value.  It's more like "units", but that's not quite right.  I'm not sure what I would call it.  "max field value" maybe, but that's just as bad.
2. using `map` isn't quite right either.  I think you're doing this a bit upside-down.  Instead of mapping `[10,60,60,24]` onto the parameter (`delta`), you're mapping `delta` onto `[10,60,60,24]`.  Or maybe I'm the one who's upside down.  Either way, I really needed to twist my neck to figure this out.  
3. `map` in the traditional map/reduce setting needs to be purely functional, but yours isn't.  By modifying `delta` inside the lambda, you confuse things.  You're assuming a sequential evaluation of the array, and you're assuming it evaluates from left-to-right.  Every execution of the lambda affects the next execution, and that forces the reader to mentally unroll the loop in their head.

Look at the `divmod` function.  I think it could really help here.

Reply:
I tried looking for a `divmod` function in the standard libraries but didn't find one, unfortunately